Source/WebCore/page/Navigation.cpp

```diff
@@ -859,14 +866,17 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
             m_ongoingNavigateEvent->finish();
             m_ongoingNavigateEvent = nullptr;
 
-            // FIXME: Fill in error information.
+            std::optional<ErrorInformation> errorInformation;
             String errorMessage;
-            if (auto* error = jsDynamicCast<JSC::ErrorInstance*>(result))
-                errorMessage = makeString("Uncaught "_s, error->sanitizedToString(protectedScriptExecutionContext()->globalObject()));
+            if (auto* errorInstance = jsDynamicCast<JSC::ErrorInstance*>(result)) {
+                errorInformation = extractErrorInformationFromErrorInstance(protectedScriptExecutionContext()->globalObject(), *errorInstance);
+                errorMessage = makeString("Uncaught "_s, errorInformation->errorTypeString, ": "_s, errorInformation->message);
+            }
+            if (!errorInformation)
+                errorInformation = { };
```

Comment:
This doesn't do what you think it does.  It checks if errorInformation is nullopt, then re-assigns it to nullopt if it is, and line 879 will still crash.

Reply:
Should be fixed.